src/components/shared/chart/Viewport.js

```diff
@@ -63,28 +90,36 @@ require('./Viewport.css');
  * viewportRight += mouseMoveDelta * unitPixel
  * viewportLeft += mouseMoveDelta * unitPixel
  **/
-export default function withChartViewport<T>(WrappedComponent: ReactClass<T>) {
-  class ChartViewport extends PureComponent {
-    props: Props;
+/**
+ * About the Flow typing:
+ *
+ * - `Props` are the props for the returned component. It means that they are the
+ *   props that the user for this component will need to specify. From the
+ *   generic definition `<Props: ViewportProps>`, they must include the
+ *   properties consumed par this HOC.
+ *
+ * - The argument, which is the augmented component (the `WrappedComponent`),
+ *   needs to accept both the `InjectedProps` and some supertype of `Props`.
+ *   A supertype of `Props` is an object will some properties of
+ *   `Props` but not all of them.
+ *   To understand what this means to Flow, we need, like sometimes, to think
+ *   backwards: from the props of `WrappedComponent`, take out `InjectedProps`,
+ *   and make it part of `Props`.
+ *
+ * So `Props` will need to hold both `ViewportProps` as said earlier and the
+ * props from `WrappedComponent` that aren't `Injectedprops`.
+ * This is exactly what we want Flow to check: that the user of this HOC
+ * properly passes all these props!
+ */
```

Comment:
I first explained in a GH comment, but then I thought you'd tell me to put it as a comment in the code ;)

Reply:
Literally was about to type "thanks for putting this in a comment!"